meson.build

```diff
@@ -0,0 +1,1449 @@
+               configuration : exportedVariables,
+               install : false)
+
+#
+# We also want to inject build-system variables into various platform-specific packaging scripts, control files and so
+# on.
+#
+# We ultimately want the generated files inside the 'packaging' subtree of the the build directory, but that is not
+# permitted by configure_file, so, that gets done by the bt script.
+#
+if host_machine.system() == 'linux'
+   #
+   # Linux: Debian Binary package control file
+   #
+   configure_file(input  : 'packaging/linux/control.in',
+                  output : 'control',
+                  configuration : exportedVariables,
+                  install : false)
+   #
+   # Linux: RPM binary package spec file
+   #
+   configure_file(input  : 'packaging/linux/rpm.spec.in',
+                  output : 'rpm.spec',
+                  configuration : exportedVariables,
+                  install : false)
+elif host_machine.system() == 'windows'
+   #
+   # Windows: NSIS installer script
+   #
+   configure_file(input  : 'packaging/windows/NsisInstallerScript.nsi.in',
+                  output : 'NsisInstallerScript.nsi',
+                  configuration : exportedVariables,
+                  install : false)
+elif host_machine.system() == 'darwin'
+   #
+   # Mac: Information Property List file
+   #
+   configure_file(input  : 'packaging/darwin/Info.plist.in',
+                  output : 'Info.plist',
+                  configuration : exportedVariables,
+                  install : false)
+endif
+
+#=======================================================================================================================
+#============================================= Generate manpage for Linux ==============================================
+#=======================================================================================================================
+if host_machine.system() == 'linux'
+   #
+   # Generate man page from markdown (which already had build system variables injected into it via meson.build)
+   # using the pandoc utility:
+   #    --verbose      = give verbose debugging output (though currently this does not have much effect)
+   #    -t man         = generate output in man format
+   #    -s             = generate a complete man page (rather than just some text in man format)
+   #    -o <filename>  = specifies the output file
+   #
+   # Note that, although man pages are frequently compressed, the exact compression mechanism is distro-specific, so
+   # Meson now considers such compression outside the scope of the build system.  (We therefore do it in the bt build
+   # tool script.)
+   #
+   # TODO: For the moment we are only producing an English man page.  It would not be huge effort to produce them for
+   # non-English locales, so we should do that at some point.
+   #
+   pandocResult = run_command(
+      'pandoc',
+      join_paths(meson.current_build_dir(), 'manpage.1.md'),
```

Comment:
If you save the `configure_file()` returned value from above, you can pass it into command lines here.

Any particular reason you used run_command() instead of something like:
```meson
man_file = configure_file(
    input: configured_md_file,
    output: projectName + '.1',
    command: ['pandoc', ......],
)
install_man(man_file)
```
?

Reply:
Ah, OK, so I can use configure_file again, but this time with a custom command.  That hadn't occurred to me.

Reply:
I hadn't noticed that `configure_file` returns a file object!